pallets/swap/src/pallet/impls.rs

```
@@ -1212,6 +1212,104 @@ impl<T: Config> Pallet<T> {
     pub fn protocol_account_id() -> T::AccountId {
         T::ProtocolId::get().into_account_truncating()
     }
+
+    /// Dissolve all LPs and clean state.
+    pub fn do_dissolve_all_liquidity_providers(netuid: NetUid) -> DispatchResult {
+        let user_lp_enabled =
+        <Self as subtensor_swap_interface::SwapHandler<T::AccountId>>::is_user_liquidity_enabled(netuid);
+
+        if SwapV3Initialized::<T>::get(netuid) {
+            // 1) Snapshot (owner, position_id).
+            struct CloseItem<A> {
+                owner: A,
+                pos_id: PositionId,
+            }
+            let mut to_close: sp_std::vec::Vec<CloseItem<T::AccountId>> = sp_std::vec::Vec::new();
+            for ((owner, pos_id), _pos) in Positions::<T>::iter_prefix((netuid,)) {
+                to_close.push(CloseItem { owner, pos_id });
+            }
+
+            let protocol_account = Self::protocol_account_id();
+
+            // Non‑protocol first
+            to_close
+                .sort_by(|a, b| (a.owner == protocol_account).cmp(&(b.owner == protocol_account)));
+
+            for CloseItem { owner, pos_id } in to_close.into_iter() {
+                let rm = Self::do_remove_liquidity(netuid, &owner, pos_id)?;
```

Would it make sense to ignore errors here since we’re in the loop? If reserves are ideal, errors will not happen, but we had a few small discrepancies in the state so liquidity removals may fail for some (larger) accounts while they will succeed for smaller ones.